test-main.js

```diff
@@ -0,0 +1,65 @@
+var allTestFiles = ['assert', 'underscore'];
+var TEST_REGEXP = /tests\/.*\.js$/i;
+
+// Automatically clenaup sinon spies and stubs.
+var realSinon = sinon;
+beforeEach(function() {
+    sinon = realSinon.sandbox.create();
+});
+afterEach(function() {
+    sinon.restore();
+    sinon = realSinon;
+});
+
+function withSettings(changes, test) {
```

Comment:
This seems like it is a pretty common pattern. There are a lot of tests using it now. This used to be handled by `mock` and I was hoping to use `sinon.stub` for it but `sinon.stub` throws if the thing you're stubbing doesn't exist.

Does this seem reasonable? Anyone have another idea for how to handle this use case?

See [tests/models.js](https://github.com/mozilla/marketplace-core-modules/blob/fa64ec6f523f44b06427275558fdfbcf303465b5/tests/models.js#L25-L29) and [tests/cache.js](https://github.com/mozilla/marketplace-core-modules/blob/fa64ec6f523f44b06427275558fdfbcf303465b5/tests/cache.js#L258-L266) for some example uses.